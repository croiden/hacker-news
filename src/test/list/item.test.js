import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'

import Item from '../../list/item'

afterEach(cleanup)

const props = {
    objectID: '17570029',
    title: 'Farewell, Google Maps',
    story_text: null,
    num_comments: 527,
    points: 1747,
    url: 'https://www.inderapotheke.de/blog/farewell-google-maps',
    author: 'ScottWRobinson',
    created_at: '2018-07-19T20:36:38.000Z',
    onUpVote: () => {},
    onHide: () => {},
}

describe('Item Component snapshot', () => {
    it('Item snapshot', () => {
        const { asFragment } = render(<Item {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })
})

describe('Testing Item Component', () => {
    it('should invoke the onUpVote on click of vote button', () => {
        const handleUpVote = jest.fn()
        const { getByLabelText } = render(<Item {...props} onUpVote={handleUpVote} />)
        expect(getByLabelText('up vote')).toBeDefined()

        fireEvent.click(getByLabelText('up vote'))
        expect(handleUpVote).toHaveBeenCalled()
        expect(handleUpVote).toHaveBeenCalledWith('17570029')
    })

    it('should invoke the onHide on click of hide button', () => {
        const handleHide = jest.fn()
        const { getByText } = render(<Item {...props} onHide={handleHide} />)
        expect(getByText('Hide')).toBeDefined()

        fireEvent.click(getByText('Hide'))
        expect(handleHide).toHaveBeenCalled()
        expect(handleHide).toHaveBeenCalledWith('17570029')
    })
})
