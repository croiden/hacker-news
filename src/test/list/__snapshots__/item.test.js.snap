// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component snapshot Item snapshot 1`] = `
<DocumentFragment>
  <li
    class="sc-AxjAm fkjhPm"
  >
    <div
      class="sc-AxirZ iRYSiM"
    >
      527
    </div>
    <div
      class="sc-AxirZ sc-AxiKw ekUlTF"
      points="1747"
    >
      1747
    </div>
    <div
      class="sc-AxirZ iRYSiM"
    >
      <button
        aria-label="up vote"
        class="sc-fzoLsD hYEUtg"
      >
        <img
          alt="up vote"
          src="/icons/triangle.svg"
        />
      </button>
    </div>
    <div
      class="sc-AxhCb ocJqW"
    >
      <div
        class="sc-AxhUy eahkvq"
      >
        <a
          href="https://www.inderapotheke.de/blog/farewell-google-maps"
        >
          Farewell, Google Maps
        </a>
      </div>
      <div
        class="sc-AxmLO hEBUuY"
      >
        <div
          class="sc-AxgMl kktBcw"
        >
          <span
            class="sc-AxheI iUKdMy"
          >
            (
            <a
              href="https://www.inderapotheke.de/blog/farewell-google-maps"
            >
              www.inderapotheke.de
            </a>
            )
          </span>
        </div>
        <div
          class="sc-AxgMl kktBcw"
        >
          <span
            class="sc-Axmtr dZwUvE"
          >
            by 
          </span>
          <label>
            ScottWRobinson
          </label>
        </div>
        <div
          class="sc-AxgMl kktBcw"
        >
          <label>
            2 years ago
          </label>
        </div>
      </div>
      <div
        class="sc-AxgMl kktBcw"
      >
        <span
          class="sc-AxheI iUKdMy"
        >
          [ 
          <button
            class="sc-fzozJi sldbA"
          >
            Hide
          </button>
           ]
        </span>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
